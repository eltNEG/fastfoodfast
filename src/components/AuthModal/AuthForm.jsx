import React from 'react';
import { Form, Button, Message } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

class AuthForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '', // eslint-disable-line react/no-unused-state
      password: '', // eslint-disable-line react/no-unused-state
      confirmPassword: '' // eslint-disable-line react/no-unused-state
    };
  }

  handleChange = (event, data) => {
    const { name, value } = data;
    this.setState({ [name]: value });
  };

  renderFormInput = (icon, name, placeholder, type) => {
    const state = this;
    return (
      <Form.Input
        fluid
        icon={icon}
        name={name}
        iconPosition="left"
        placeholder={placeholder}
        type={type}
        onChange={this.handleChange}
        value={state[name]}
      />
    );
  };

  renderRegisterForm = () => (
    <Form onSubmit={this.handleSubmit}>
      <h2>Register a new account</h2>
      {this.renderFormInput('user', 'username', 'Username', 'text')}
      {this.renderFormInput('lock', 'password', 'Password', 'password')}
      {this.renderFormInput('lock', 'confirmPassword', 'Confirm Password', 'password')}
      <Button as="div" fluid size="large">
        Register
      </Button>
      <Message>
        Already have an account?
        {' '}
        <Link to="/login">Login</Link>
      </Message>
    </Form>
  );

  renderLoginForm = () => (
    <Form>
      <h2>Log-in to your account</h2>
      {this.renderFormInput('user', 'username', 'Username', 'text')}
      {this.renderFormInput('lock', 'password', 'Password', 'password')}
      <Button as="div" fluid size="large">
        Login
      </Button>
      <Message>
        Do not have an account?
        {' '}
        <Link to="/register">Register</Link>
      </Message>
    </Form>
  );

  render() {
    const { formType } = this.props;
    switch (formType) {
      case 'Register':
        return this.renderRegisterForm();
      case 'Login':
        return this.renderLoginForm();
      default:
        return this.renderRegisterForm();
    }
  }
}

AuthForm.propTypes = {
  formType: PropTypes.string.isRequired
};

export default AuthForm;
