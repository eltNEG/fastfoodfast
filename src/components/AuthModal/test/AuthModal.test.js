import React from 'react';
import { shallow } from 'enzyme';
import { AuthModalComponent } from '..';
import AuthForm from '../AuthForm';

describe('AuthModal component', () => {
  it('should match AuthModal.jsx snapshot', () => {
    const wrapper = shallow(
      <AuthModalComponent
        text="register"
        history={{}}
        children={<div />}
        location={{ pathname: '/register' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should match AuthModal.jsx snapshot', () => {
    const wrapper = shallow(
      <AuthModalComponent
        text="register"
        history={{}}
        children={<div />}
        location={{ pathname: '/login' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should set push to history', () => {
    const history = {
      push: jest.fn()
    };
    const wrapper = shallow(
      <AuthModalComponent
        text="register"
        history={history}
        children={<div />}
        location={{ pathname: '/login' }}
      />
    );
    wrapper.instance().handleClose();
    expect(history.push).toBeCalledTimes(1);
  });
  it('should match AuthForm.jsx snapshot', () => {
    const wrapper = shallow(<AuthForm formType="Register" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match AuthForm.jsx snapshot', () => {
    const wrapper = shallow(<AuthForm formType="Login" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match AuthForm.jsx snapshot', () => {
    const wrapper = shallow(<AuthForm formType="Log" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should set state', () => {
    const wrapper = shallow(<AuthForm formType="Log" />);
    wrapper.instance().handleChange({}, { name: 'username', value: 'user' });
    expect(wrapper.state().username).toBe('user');
  });
});
