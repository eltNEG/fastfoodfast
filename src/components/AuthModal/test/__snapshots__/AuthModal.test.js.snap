// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthModal component should match AuthForm.jsx snapshot 1`] = `
<Form
  as="form"
>
  <h2>
    Register a new account
  </h2>
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    icon="user"
    iconPosition="left"
    name="username"
    onChange={[Function]}
    placeholder="Username"
    type="text"
  />
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    icon="lock"
    iconPosition="left"
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
  />
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    icon="lock"
    iconPosition="left"
    name="confirmPassword"
    onChange={[Function]}
    placeholder="Confirm Password"
    type="password"
  />
  <Button
    as="div"
    fluid={true}
    size="large"
  >
    Register
  </Button>
  <Message>
    Already have an account?
     
    <Link
      replace={false}
      to="/login"
    >
      Login
    </Link>
  </Message>
</Form>
`;

exports[`AuthModal component should match AuthForm.jsx snapshot 2`] = `
<Form
  as="form"
>
  <h2>
    Log-in to your account
  </h2>
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    icon="user"
    iconPosition="left"
    name="username"
    onChange={[Function]}
    placeholder="Username"
    type="text"
  />
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    icon="lock"
    iconPosition="left"
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
  />
  <Button
    as="div"
    fluid={true}
    size="large"
  >
    Login
  </Button>
  <Message>
    Do not have an account?
     
    <Link
      replace={false}
      to="/register"
    >
      Register
    </Link>
  </Message>
</Form>
`;

exports[`AuthModal component should match AuthForm.jsx snapshot 3`] = `
<Form
  as="form"
>
  <h2>
    Register a new account
  </h2>
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    icon="user"
    iconPosition="left"
    name="username"
    onChange={[Function]}
    placeholder="Username"
    type="text"
  />
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    icon="lock"
    iconPosition="left"
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
  />
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    icon="lock"
    iconPosition="left"
    name="confirmPassword"
    onChange={[Function]}
    placeholder="Confirm Password"
    type="password"
  />
  <Button
    as="div"
    fluid={true}
    size="large"
  >
    Register
  </Button>
  <Message>
    Already have an account?
     
    <Link
      replace={false}
      to="/login"
    >
      Login
    </Link>
  </Message>
</Form>
`;

exports[`AuthModal component should match AuthModal.jsx snapshot 1`] = `
<Modal
  basic={true}
  centered={true}
  className="auth-modal"
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
  onClose={[Function]}
  open={true}
  size="small"
  trigger={
    <div>
      register
    </div>
  }
>
  <div />
</Modal>
`;

exports[`AuthModal component should match AuthModal.jsx snapshot 2`] = `
<Modal
  basic={true}
  centered={true}
  className="auth-modal"
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
  onClose={[Function]}
  open={false}
  size="small"
  trigger={
    <div>
      register
    </div>
  }
>
  <div />
</Modal>
`;
